programs/perpetuals/src/instructions/add_collateral.rs

```diff
@@ -143,11 +147,52 @@ pub fn add_collateral(ctx: Context<AddCollateral>, params: &AddCollateralParams)
     msg!("Collateral added in USD: {}", collateral_usd);
 
     // update existing position
+    // changing collateral amount will change position leverage and affect pnl and interest calculations,
+    // we need to save current pnl and interest as unrealized and adjust the position price
```

Comment:
why will changing collateral affect pnl calculation? it will affect the interest calculation but I don't think we should calculate the interest by `size - collateral`

Reply:
i think the calculation of pnl is incorrect, it should be pnl = (size / position_price) * price_diff right?

for borrow interest, we should charge interest on the full size amount

Reply:
alright, I made interest to be charged on the full size.
where did you get that formula for pnl? In that case it wouldn't even depend on the leverage

Reply:
gmx calculate this way also, your formulae will correct when the collateral is 1 SOL.

Reply:
yeah, you are right, let me close this and we work on #22 since you have done similar work over there